### ISSUE TITLE
Build under Fedora 28, PHP 7.2.7,  fails

### ISSUE THREAD

**Issue by mrychlik** [chobie/php-sundown#45]
/bin/sh /home/marek/Sundown/php-sundown/libtool --mode=compile cc  -I. -I/home/marek/Sundown/php-sundown -DPHP_ATOM_INC -I/home/marek/Sundown/php-sundown/include -I/home/marek/Sundown/php-sundown/main -I/home/marek/Sundown/php-sundown -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -I/home/marek/Sundown/php-sundown/sundown/src -I/home/marek/Sundown/php-sundown/sundown/html  -DHAVE_CONFIG_H  -Wunused-variable -Wpointer-sign -Wimplicit-function-declaration -Winline -Wunused-macros -Wredundant-decls -Wstrict-aliasing=2 -Wswitch-enum -Wdeclaration-after-statement   -c /home/marek/Sundown/php-sundown/php_sundown.c -o php_sundown.lo 
libtool: compile:  cc -I. -I/home/marek/Sundown/php-sundown -DPHP_ATOM_INC -I/home/marek/Sundown/php-sundown/include -I/home/marek/Sundown/php-sundown/main -I/home/marek/Sundown/php-sundown -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -I/home/marek/Sundown/php-sundown/sundown/src -I/home/marek/Sundown/php-sundown/sundown/html -DHAVE_CONFIG_H -Wunused-variable -Wpointer-sign -Wimplicit-function-declaration -Winline -Wunused-macros -Wredundant-decls -Wstrict-aliasing=2 -Wswitch-enum -Wdeclaration-after-statement -c /home/marek/Sundown/php-sundown/php_sundown.c  -fPIC -DPIC -o .libs/php_sundown.o
In file included from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/php/main/php.h:336:15: warning: redundant redeclaration of ‘environ’ [-Wredundant-decls]
 extern char **environ;
               ^~~~~~~
In file included from /usr/include/php/main/php.h:233,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/unistd.h:545:15: note: previous declaration of ‘environ’ was here
 extern char **environ;
               ^~~~~~~
In file included from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/home/marek/Sundown/php-sundown/php_sundown.h: In function ‘php_sundown_has_ext’:
/home/marek/Sundown/php-sundown/php_sundown.h:142:28: warning: passing argument 2 of ‘zend_hash_find’ from incompatible pointer type [-Wincompatible-pointer-types]
  if (zend_hash_find(table, name, length, (void **)&data) == SUCCESS) {
                            ^~~~
In file included from /usr/include/php/Zend/zend.h:34,
                 from /usr/include/php/main/php.h:35,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/php/Zend/zend_hash.h:168:30: note: expected ‘zend_string *’ {aka ‘struct _zend_string *’} but argument is of type ‘const char *’
 ZEND_API zval* ZEND_FASTCALL zend_hash_find(const HashTable *ht, zend_string *key);
                              ^~~~~~~~~~~~~~
In file included from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/home/marek/Sundown/php-sundown/php_sundown.h:142:6: error: too many arguments to function ‘zend_hash_find’
  if (zend_hash_find(table, name, length, (void **)&data) == SUCCESS) {
      ^~~~~~~~~~~~~~
In file included from /usr/include/php/Zend/zend.h:34,
                 from /usr/include/php/main/php.h:35,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/php/Zend/zend_hash.h:168:30: note: declared here
 ZEND_API zval* ZEND_FASTCALL zend_hash_find(const HashTable *ht, zend_string *key);
                              ^~~~~~~~~~~~~~
In file included from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/home/marek/Sundown/php-sundown/php_sundown.h:143:7: warning: implicit declaration of function ‘Z_TYPE_PP’; did you mean ‘Z_TYPE_P’? [-Wimplicit-function-declaration]
   if (Z_TYPE_PP(data) != IS_BOOL) {
       ^~~~~~~~~
       Z_TYPE_P
/home/marek/Sundown/php-sundown/php_sundown.h:143:26: error: ‘IS_BOOL’ undeclared (first use in this function); did you mean ‘INI_BOOL’?
   if (Z_TYPE_PP(data) != IS_BOOL) {
                          ^~~~~~~
                          INI_BOOL
/home/marek/Sundown/php-sundown/php_sundown.h:143:26: note: each undeclared identifier is reported only once for each function it appears in
/home/marek/Sundown/php-sundown/php_sundown.h:147:7: warning: implicit declaration of function ‘Z_BVAL_PP’; did you mean ‘Z_DVAL_P’? [-Wimplicit-function-declaration]
   if (Z_BVAL_PP(data)) {
       ^~~~~~~~~
       Z_DVAL_P
/home/marek/Sundown/php-sundown/php_sundown.h: At top level:
/home/marek/Sundown/php-sundown/php_sundown.h:157:117: error: unknown type name ‘zend_uint’; did you mean ‘zend_llist’?
 unction_table, zval **object_pp, zval *function_name, zval *retval_ptr, zend_uint param_count, ...)
                                                                         ^~~~~~~~~
                                                                         zend_llist
In file included from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/home/marek/Sundown/php-sundown/php_sundown.h: In function ‘buf2str’:
/home/marek/Sundown/php-sundown/php_sundown.h:193:2: warning: implicit declaration of function ‘MAKE_STD_ZVAL’ [-Wimplicit-function-declaration]
  MAKE_STD_ZVAL(str);
  ^~~~~~~~~~~~~
/home/marek/Sundown/php-sundown/php_sundown.h:197:54: error: macro "ZVAL_STRINGL" passed 4 arguments, but takes just 3
   ZVAL_STRINGL(str, (char *)text->data, text->size, 1);
                                                      ^
/home/marek/Sundown/php-sundown/php_sundown.h:197:3: error: ‘ZVAL_STRINGL’ undeclared (first use in this function); did you mean ‘ZEND_STR_NULL’?
   ZVAL_STRINGL(str, (char *)text->data, text->size, 1);
   ^~~~~~~~~~~~
   ZEND_STR_NULL
/home/marek/Sundown/php-sundown/php_sundown.h: In function ‘char2str’:
/home/marek/Sundown/php-sundown/php_sundown.h:207:26: error: macro "ZVAL_STRING" passed 3 arguments, but takes just 2
  ZVAL_STRING(str, text, 1);
                          ^
/home/marek/Sundown/php-sundown/php_sundown.h:207:2: error: ‘ZVAL_STRING’ undeclared (first use in this function); did you mean ‘ZEND_STR_LINE’?
  ZVAL_STRING(str, text, 1);
  ^~~~~~~~~~~
  ZEND_STR_LINE
/home/marek/Sundown/php-sundown/php_sundown.c: In function ‘zim_sundown___construct’:
/home/marek/Sundown/php-sundown/php_sundown.c:60:2: error: too many arguments to function ‘add_property_string_ex’
  add_property_string_ex(getThis(), ZEND_STRS("data"), buffer, 1 TSRMLS_CC);
  ^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/main/php.h:39,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/php/Zend/zend_API.h:474:14: note: declared here
 ZEND_API int add_property_string_ex(zval *arg, const char *key, size_t key_len, const char *str);
              ^~~~~~~~~~~~~~~~~~~~~~
/home/marek/Sundown/php-sundown/php_sundown.c:66:3: warning: implicit declaration of function ‘ALLOC_INIT_ZVAL’ [-Wimplicit-function-declaration]
   ALLOC_INIT_ZVAL(c_extensions);
   ^~~~~~~~~~~~~~~
/home/marek/Sundown/php-sundown/php_sundown.c: In function ‘zim_sundown___destruct’:
/home/marek/Sundown/php-sundown/php_sundown.c:79:15: error: too few arguments to function ‘zend_read_property’
  extensions = zend_read_property(sundown_class_entry, getThis(), ZEND_STRS("extensions")-1, 0 TSRMLS_CC);
               ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/main/php.h:39,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/php/Zend/zend_API.h:373:16: note: declared here
 ZEND_API zval *zend_read_property(zend_class_entry *scope, zval *object, const char *name, size_t name_length, zend_bool silent, zval *rv);
                ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/Zend/zend.h:37,
                 from /usr/include/php/main/php.h:35,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/home/marek/Sundown/php-sundown/php_sundown.c:80:16: warning: passing argument 1 of ‘_zval_ptr_dtor’ from incompatible pointer type [-Wincompatible-pointer-types]
  zval_ptr_dtor(&extensions);
/usr/include/php/Zend/zend_variables.h:87:49: note: in definition of macro ‘zval_ptr_dtor’
 #define zval_ptr_dtor(zval_ptr) _zval_ptr_dtor((zval_ptr) ZEND_FILE_LINE_CC)
                                                 ^~~~~~~~
/usr/include/php/Zend/zend_variables.h:79:36: note: expected ‘zval *’ {aka ‘struct _zval_struct *’} but argument is of type ‘zval **’ {aka ‘struct _zval_struct **’}
 ZEND_API void _zval_ptr_dtor(zval *zval_ptr ZEND_FILE_LINE_DC);
                              ~~~~~~^~~~~~~~
In file included from /usr/include/php/Zend/zend.h:29,
                 from /usr/include/php/main/php.h:35,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/home/marek/Sundown/php-sundown/php_sundown.c: In function ‘zim_sundown_hasExtension’:
/home/marek/Sundown/php-sundown/php_sundown.c:121:15: error: too few arguments to function ‘zend_read_property’
  if (Z_TYPE_P(zend_read_property(sundown_class_entry, getThis(), ZEND_STRS("extensions")-1, 0 TSRMLS_CC)) != IS_NULL) {
               ^~~~~~~~~~~~~~~~~~
/usr/include/php/Zend/zend_types.h:399:41: note: in definition of macro ‘Z_TYPE’
 #define Z_TYPE(zval)    zval_get_type(&(zval))
                                         ^~~~
/home/marek/Sundown/php-sundown/php_sundown.c:121:6: note: in expansion of macro ‘Z_TYPE_P’
  if (Z_TYPE_P(zend_read_property(sundown_class_entry, getThis(), ZEND_STRS("extensions")-1, 0 TSRMLS_CC)) != IS_NULL) {
      ^~~~~~~~
In file included from /usr/include/php/main/php.h:39,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/php/Zend/zend_API.h:373:16: note: declared here
 ZEND_API zval *zend_read_property(zend_class_entry *scope, zval *object, const char *name, size_t name_length, zend_bool silent, zval *rv);
                ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/Zend/zend.h:29,
                 from /usr/include/php/main/php.h:35,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/home/marek/Sundown/php-sundown/php_sundown.c:122:22: error: too few arguments to function ‘zend_read_property’
   table = Z_ARRVAL_P(zend_read_property(sundown_class_entry, getThis(), ZEND_STRS("extensions")-1, 0 TSRMLS_CC));
                      ^~~~~~~~~~~~~~~~~~
/usr/include/php/Zend/zend_types.h:590:26: note: in definition of macro ‘Z_ARR’
 #define Z_ARR(zval)     (zval).value.arr
                          ^~~~
/usr/include/php/Zend/zend_types.h:594:30: note: in expansion of macro ‘Z_ARRVAL’
 #define Z_ARRVAL_P(zval_p)   Z_ARRVAL(*(zval_p))
                              ^~~~~~~~
/home/marek/Sundown/php-sundown/php_sundown.c:122:11: note: in expansion of macro ‘Z_ARRVAL_P’
   table = Z_ARRVAL_P(zend_read_property(sundown_class_entry, getThis(), ZEND_STRS("extensions")-1, 0 TSRMLS_CC));
           ^~~~~~~~~~
In file included from /usr/include/php/main/php.h:39,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/php/Zend/zend_API.h:373:16: note: declared here
 ZEND_API zval *zend_read_property(zend_class_entry *scope, zval *object, const char *name, size_t name_length, zend_bool silent, zval *rv);
                ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/Zend/zend.h:29,
                 from /usr/include/php/main/php.h:35,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/home/marek/Sundown/php-sundown/php_sundown.c: In function ‘zim_sundown_hasRenderFlag’:
/home/marek/Sundown/php-sundown/php_sundown.c:140:15: error: too few arguments to function ‘zend_read_property’
  if (Z_TYPE_P(zend_read_property(sundown_class_entry, getThis(), ZEND_STRS("extensions")-1, 0 TSRMLS_CC)) != IS_NULL) {
               ^~~~~~~~~~~~~~~~~~
/usr/include/php/Zend/zend_types.h:399:41: note: in definition of macro ‘Z_TYPE’
 #define Z_TYPE(zval)    zval_get_type(&(zval))
                                         ^~~~
/home/marek/Sundown/php-sundown/php_sundown.c:140:6: note: in expansion of macro ‘Z_TYPE_P’
  if (Z_TYPE_P(zend_read_property(sundown_class_entry, getThis(), ZEND_STRS("extensions")-1, 0 TSRMLS_CC)) != IS_NULL) {
      ^~~~~~~~
In file included from /usr/include/php/main/php.h:39,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/php/Zend/zend_API.h:373:16: note: declared here
 ZEND_API zval *zend_read_property(zend_class_entry *scope, zval *object, const char *name, size_t name_length, zend_bool silent, zval *rv);
                ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/Zend/zend.h:29,
                 from /usr/include/php/main/php.h:35,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/home/marek/Sundown/php-sundown/php_sundown.c:141:22: error: too few arguments to function ‘zend_read_property’
   table = Z_ARRVAL_P(zend_read_property(sundown_class_entry, getThis(), ZEND_STRS("extensions")-1, 0 TSRMLS_CC));
                      ^~~~~~~~~~~~~~~~~~
/usr/include/php/Zend/zend_types.h:590:26: note: in definition of macro ‘Z_ARR’
 #define Z_ARR(zval)     (zval).value.arr
                          ^~~~
/usr/include/php/Zend/zend_types.h:594:30: note: in expansion of macro ‘Z_ARRVAL’
 #define Z_ARRVAL_P(zval_p)   Z_ARRVAL(*(zval_p))
                              ^~~~~~~~
/home/marek/Sundown/php-sundown/php_sundown.c:141:11: note: in expansion of macro ‘Z_ARRVAL_P’
   table = Z_ARRVAL_P(zend_read_property(sundown_class_entry, getThis(), ZEND_STRS("extensions")-1, 0 TSRMLS_CC));
           ^~~~~~~~~~
In file included from /usr/include/php/main/php.h:39,
                 from /home/marek/Sundown/php-sundown/php_sundown.h:12,
                 from /home/marek/Sundown/php-sundown/php_sundown.c:20:
/usr/include/php/Zend/zend_API.h:373:16: note: declared here
 ZEND_API zval *zend_read_property(zend_class_entry *scope, zval *object, const char *name, size_t name_length, zend_bool silent, zval *rv);
                ^~~~~~~~~~~~~~~~~~
make: *** [Makefile:194: php_sundown.lo] Error 1

**Comment by j5lx** [comment]
Duplicate of #42
